src/components/Home.js

```diff
@@ -11,22 +12,53 @@ const Home = () => {
     url += '&app_id=' + ID;
     url += '&app_key=' + KEY;
 
+    const [recipes, setRecipes] = useState([]);
+    const [recipeTitles, setRecipeTitles] = useState([]);
+    const [recipeImages, setRecipeImages] = useState([]);
+    const [recipeIngredients, setRecipeIngredients] = useState([]);
+    const [recipeLinks, setRecipeLinks] = useState([]);
+    const [recipeIndexes, setRecipeIndexes] = useState([]);
+    const [cards, setCards] = useState(false);
+
     const getRandomRecipes = async () => {
         const response = await fetch(url);
         const body = await response.json();
-        console.log(body);
+        const recipeData = body.hits.map(hit => hit.recipe);
+
+        setRecipes(recipeData);
+        setRecipeTitles(recipeData.map(recipe => recipe.label));
+        setRecipeImages(recipeData.map(recipe => recipe.images.REGULAR));
+        setRecipeIngredients(recipeData.map(recipe => recipe.ingredientLines));
+        setRecipeLinks(recipeData.map(recipe => recipe.shareAs));
+        setRecipeIndexes(generateIndexes(recipeData.length));
+        
+        showCards();
+    }
+
+    const generateIndexes = (length) => {
+        const randomNumbers = new Set();
+        while (randomNumbers.size < 3) {
```

Comment:
Are you good with just 3 recipies being displayed? I also think that maybe if cards are displayed on the screen then 'Generate Recipies' should be changed to 'Refresh'

Reply:
I think 3 recipes is good because the point is to be quick (and not overload them with options). Plus, they can always regenerate the recipes if they don't like any of the options. I have changed the text on the button when the cards are showing.